dawn/src/dawn/IIR/Extents.h

```diff
@@ -85,27 +85,27 @@ class HorizontalExtentImpl {
   virtual ~HorizontalExtentImpl() = default;
 
   HorizontalExtentImpl& operator+=(HorizontalExtentImpl const& other) {
-    add_impl(other);
+    addImpl(other);
     return *this;
   }
-  std::unique_ptr<HorizontalExtentImpl> clone() const { return clone_impl(); }
+  std::unique_ptr<HorizontalExtentImpl> clone() const { return cloneImpl(); }
 
-  void merge(HorizontalExtentImpl const& other) { merge_impl(other); }
-  void merge(ast::HorizontalOffset const& other) { merge_impl(other); }
-  bool equals(HorizontalExtentImpl const& other) const { return equals_impl(other); }
-  bool isPointwise() const { return isPointwise_impl(); }
+  void merge(HorizontalExtentImpl const& other) { mergeImpl(other); }
+  void addCenter() { addCenterImpl(); }
```

Comment:
The problem is in the implementation of `merge`. You can do:
```
HorizontalExtent my_extent{cartesian, 1, 2, 3, 4, 5, 6};
HorizontalExtent zero_extent{};
my_extent->merge(zero_extent);
```
This adds zero to `my_extent`. For `my_extent`, `impl_` is a `CartesianExtent`, for `zero_extent`, `impl_` is `nullptr`.

`merge` now needs to merge those two extents. `HorizontalExtentImpl` has `HorizontalExtentImpl::merge(HorizontalExtentImpl const& other);`, but this I cannot call because the RHS does not exist.

Another option would be to have `HorizontalExtentImpl::merge(HorizontalExtentImpl const*)`, but then we would just duplicate the logic into the `merge` functions of `CartesianExtent` and `UnstructuredExtent`.

Reply:
Ok.